drivers/stepper/adi_tmc/adi_tmc5041_stepper_controller.c

```diff
@@ -103,20 +106,30 @@ static void calculate_velocity_from_hz_to_fclk(const struct device *dev, const u
 		velocity_hz, *velocity_fclk);
 }
 
-static void set_async_signal(const struct device *dev, struct k_poll_signal *async)
+#ifdef CONFIG_STEPPER_ASYNC_API
+
+static int tmc5041_stepper_set_callback(const struct device *dev, stepper_callback_t  callback,
```

Comment:
@bjarki-andreasen does this seem fine to you? The end user could attach a callback(probably distribute the events in the system using a zbus channel) and also "wait(if required)" with this particular function.